File: app/src/main/java/ru/gsench/githubusers/presentation/view/UserView.java
package ru.gsench.githubusers.presentation.view;

import java.util.ArrayList;

import ru.gsench.githubusers.domain.github_repo.GitHubRepository;
import ru.gsench.githubusers.domain.github_repo.GitHubUser;
import ru.gsench.githubusers.domain.interactor.UserModel;

/**
 * Created by grish on 09.05.2017.
 */

public interface UserView {
    public void init();
    public void setUser(GitHubUser param);
    public void setUser(UserModel userShort);
    public void setRepositories(ArrayList<GitHubRepository> param);
    public void closeView();
    public void showLoadingError();
    public void showParseError();
    public void showUnexpectedError();
    public void showLoading();
    public void hideLoading();
}

File: app/src/main/java/ru/gsench/githubusers/presentation/view/UserView.kt
package ru.gsench.githubusers.presentation.view

import ru.gsench.githubusers.domain.github_repo.GitHubUser
import ru.gsench.githubusers.domain.interactor.UserModel

/**
 * Created by grish on 09.05.2017.
 */

interface UserView {
    fun init()
    fun showWithAnimation()
    fun setUser(param: GitHubUser)
    fun setUser(userShort: UserModel)
    fun notifyReposUpdate(offset: Int, count: Int)
    fun closeView()
    fun showLoadingError()
    fun showParseError()
    fun showUnexpectedError()
    fun showUserLoading()
    fun hideUserLoading()
    fun showRepoLoading()
    fun hideRepoLoading()
}
